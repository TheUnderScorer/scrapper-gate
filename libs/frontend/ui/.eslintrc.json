{
  "extends": ["plugin:@nrwl/nx/react", "../../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "libs/frontend/ui/tsconfig.*?.json",
          "libs/frontend/ui/.storybook/tsconfig.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
