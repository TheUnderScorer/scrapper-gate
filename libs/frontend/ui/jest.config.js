module.exports = {
  displayName: 'frontend-ui',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/frontend/ui',
  setupFilesAfterEnv: [`${process.cwd()}/tests/setupFrontendTests.ts`],
};
