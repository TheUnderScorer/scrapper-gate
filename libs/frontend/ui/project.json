{
  "root": "libs/frontend/ui",
  "sourceRoot": "libs/frontend/ui/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/frontend/ui/**/*.{ts,tsx,js,jsx}"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/frontend/ui"],
      "options": {
        "jestConfig": "libs/frontend/ui/jest.config.js",
        "passWithNoTests": true
      }
    },
    "storybook": {
      "executor": "@nrwl/storybook:storybook",
      "options": {
        "uiFramework": "@storybook/react",
        "port": 4400,
        "config": {
          "configFolder": "libs/frontend/ui/.storybook"
        }
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "build-storybook": {
      "executor": "@nrwl/storybook:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "uiFramework": "@storybook/react",
        "outputPath": "dist/storybook/frontend-ui",
        "config": {
          "configFolder": "libs/frontend/ui/.storybook"
        }
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    }
  },
  "tags": []
}
