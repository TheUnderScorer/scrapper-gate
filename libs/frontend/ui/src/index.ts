export * from './atoms/Centered/Centered';
export * from './atoms/Emoji/Emoji';
export * from './molecules/ErrorAlert/ErrorAlert';
export * from './molecules/SimpleDialog/SimpleDialog';
export * from './molecules/Scoped/Scoped';
export * from './styles/styles.types';
export * from './styles/highlight';
export * from './styles/Styles';
export * from './molecules/Layout/Layout';
export * from './organisms/ControlledList/ControlledList';
export * from './organisms/FlowBuilder';
export * from './assets/getAsset';
export * from './assets/useAsset';
export * from './assets/assets.types';
export * from './atoms/Image/Image';
export * from './molecules/QueryDrawer/QueryDrawer';
export * from './molecules/Dropdown/Dropdown';
export * from './atoms/TooltipText/TooltipText';
export * from './molecules/TextWithKeyHint/TextWithKeyHint';
export * from './molecules/UndoButtons/UndoButtons';
export * from './atoms/Buttons/Buttons';
export * from './molecules/ResizablePanel/ResizablePanel';
export * from './molecules/ResizablePanel/ResizablePanel.types';
export * from './molecules/FilterTextField/FilterTextField';
export * from './molecules/ContextMenu/ContextMenu';
export * from './molecules/ContextMenu/ContextMenu.types';
export * from './organisms/HtmlElementPicker/HtmlElementPicker';
export * from './organisms/HtmlElementPicker/HtmlElementPicker.types';
export * from './molecules/Incrementator/Incrementator';
export * from './styles/reactFlowBuilder';
export * from './styles/hiddenNumericArrows';
export * from './molecules/SnackbarActions/SnackbarActions';
export * from './atoms/LightTooltip/LightTooltip';
export * from './atoms/Highlight/Highlight';
export * from './molecules/InformationBox/InformationBox';
