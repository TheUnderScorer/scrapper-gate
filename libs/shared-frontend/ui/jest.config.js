module.exports = {
  displayName: 'shared-frontend-ui',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/shared-frontend/ui',
};
