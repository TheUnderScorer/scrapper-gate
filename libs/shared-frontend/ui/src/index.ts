export * from './atoms/Centered/Centered';
export * from './atoms/Emoji/Emoji';
export * from './molecules/ErrorAlert/ErrorAlert';
