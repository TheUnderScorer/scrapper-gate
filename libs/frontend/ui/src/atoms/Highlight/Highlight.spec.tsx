import { Highlight } from '@scrapper-gate/frontend/ui';
import { render } from '@testing-library/react';

describe('<Highlight />', () => {
  it('should highlight text', () => {
    const cmp = render(<Highlight text="Cool text" value="text" />);

    expect(cmp.container.outerHTML).toMatchSnapshot();
  });

  it('should highlight array of text', () => {
    const cmp = render(
      <Highlight
        text="Lorem ipsum ipsum Lorem test"
        value={['ipsum', 'test']}
      />
    );

    expect(cmp.container.outerHTML).toMatchSnapshot();
  });
});
