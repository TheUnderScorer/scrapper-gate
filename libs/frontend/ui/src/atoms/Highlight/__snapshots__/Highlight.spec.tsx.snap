// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should highlight array of text 1`] = `"<div><span>Lorem </span><span style=\\"font-weight: 700;\\">ipsum</span><span> </span><span style=\\"font-weight: 700;\\">ipsum</span><span> Lorem </span><span style=\\"font-weight: 700;\\">test</span></div>"`;

exports[`<Highlight /> should highlight text 1`] = `"<div><span>Cool </span><span style=\\"font-weight: 700;\\">text</span></div>"`;
